src/UI/Implementation/Component/Button/Renderer.php

```diff
@@ -45,18 +50,24 @@ protected function renderButton(Component\Button\Button $component, RendererInte
 		}
 
 		$tpl = $this->getTemplate($tpl_name, true, true);
+
 		$action = $component->getAction();
-		// The action is always put in the data-action attribute to have it available
-		// on the client side, even if it is not available on rendering.
-		$tpl->setVariable("ACTION", $action);
+		if ($action) {
```

Comment:
For my curiosity: This seems to imply that the invariant that buttons have actions is broken somehow for the types of buttons that are rendered here. How?

Reply:
oh, your right, actions are mandatory. sorry, just changed back that